thunder/core/codeutils.py

```diff
@@ -243,12 +243,14 @@ def prettyprint(
         # NOTE: The `class` packagename1_MyContainer will present in `import_ctx` and passed to the compiled function.
         # This is taken care of by function `to_printable`.
         name = _generate_dataclass_class_name(x)
-        instance_repr = str(x)
-        parens_idx = instance_repr.find("(")
-        call_repr = instance_repr[
-            parens_idx:
-        ]  # only keep the construction part of the repr (as we will use our generated name)
-        return m(f"{name + call_repr}")
+        call_repr_str = ""
```

Comment:
I think it would be a tad neater to build a list and then join.
a loop of `+=` on strings isn't as bad as it sometimes is made out to be, but usually using a list to accumulate is prefered.

Reply:
Thanks for the feedback. Let me update this!